src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/stdx/string_view.hpp

```diff
@@ -14,92 +14,489 @@
+     *
+     * @param other The "right hand" operand of the comparison
+     * @retval 0 If *this == other
+     * @retval n : n < 0 if *this is "less than" other.
+     * @retval n : n > 0 if *this is "greater than" other.
+     */
+    constexpr int compare(self_type other) const noexcept {
+        // Another level of indirection to support restricted C++11 constexpr
+        return _compare2(Traits::compare(data(), other.data(), (std::min)(size(), other.size())),
+                         other);
+    }
+
+    /**
+     * @brief Compare a substring of *this with `other`
+     *
+     * @returns substr(po1, count1).compare(other)
+     */
+    constexpr int compare(size_type pos1, size_type count1, self_type other) const noexcept {
+        return substr(pos1, count1).compare(other);
+    }
+
+    /**
+     * @brief Compare a substring of *this with a substring of `other`
+     *
+     * @returns substr(pos1, count1).compare(other.substr(pos2, count2))
+     */
+    constexpr int compare(size_type pos1,
+                          size_type count1,
+                          self_type other,
+                          size_type pos2,
+                          size_type count2) const noexcept {
+        return substr(pos1, count1).compare(other.substr(pos2, count2));
+    }
+
+    /**
+     * @brief Compare a substring of *this with a string viewed through the given pointer+size
+     *
+     * @returns substr(pos1, count1).compare(basic_string_view(str, count2))
+     */
+    constexpr int compare(size_type pos1,
+                          size_type count1,
+                          const_pointer str,
+                          size_type count2) const noexcept {
+        return substr(pos1, count1).compare(self_type(str, count2));
+    }
+
+    /**
+     * @brief Find the zero-based offset of the left-most occurrence of the given infix,
+     * starting with pos. If infix does not occur, returns npos.
+     */
+    bsoncxx_cxx14_constexpr size_type find(self_type infix, size_type pos = 0) const noexcept {
+        self_type sub = this->substr((std::min)(pos, size()));
+        bsoncxx::detail::subrange<iterator> found = bsoncxx::detail::search(sub, infix);
+        if (bsoncxx::detail::distance(found) != static_cast<difference_type>(infix.size())) {
+            return npos;
+        }
+        return static_cast<size_type>(found.begin() - begin());
+    }
+
+    /**
+     * @brief Find the zero-based offset of the right-most occurrence of the given infix,
+     * starting with (and including) pos. If infix does not occur, returns npos.
+     */
+    bsoncxx_cxx14_constexpr size_type rfind(self_type infix, size_type pos = npos) const noexcept {
+        // Calc the endpos where searching should begin, which includes the infix size
+        const size_type endpos = pos != npos ? pos + infix.size() : pos;
+        self_type searched = this->substr(0, endpos);
+        bsoncxx::detail::reversed_t<self_type> found =
```

Comment:
Implement this and functions below in terms of C++11 `<algorithm>` instead of C++20 Ranges.

e.g. this function is just a `std::search()` with reverse iterators:

```cpp
if (self->empty() && infix.empty()) {
  return 0u; // Special case that `std::search` doesn't handle.
}

const auto sub = self.substr((std::min)(pos, this->size()));
const auto riter = std::search(sub.rbegin(), sub.rend(), infix.rbegin(), infix.rend());

if (riter == sub.rend()) {
  return npos;
}

return static_cast<std::size_t>(
  std::distance(
    this->begin(),
    std::prev(
      riter.base(),
      static_cast<std::ptrdiff_t>(infix.size())
    )
  )
);
```